packages/grafana-data/src/transformations/transformers/sort.ts

```diff
@@ -0,0 +1,60 @@
+import { cloneDeep } from 'lodash';
+import { DataTransformerID } from './ids';
+import { DataTransformerInfo } from '../../types/transformations';
+import { DataFrame } from '../..';
+import { Field } from '../../types';
+
+export interface SortFieldsTransformerOptions {
+  indexByName: Record<string, number>;
+}
+
+export const sortFieldsTransformer: DataTransformerInfo<SortFieldsTransformerOptions> = {
+  id: DataTransformerID.sort,
+  name: 'Sort fields by name',
+  description: 'sort fields based on configuration given by user',
+  defaultOptions: {
+    indexByName: {},
+  },
+
+  /**
+   * Return a modified copy of the series.  If the transform is not or should not
+   * be applied, just return the input series
+   */
+  transformer: (options: SortFieldsTransformerOptions) => {
+    const indexByName = cloneDeep(options.indexByName);
+    const fieldsSorter = createFieldsSorter(indexByName);
+
+    return (data: DataFrame[]) => {
+      if (!Array.isArray(data) || data.length === 0) {
+        return data;
+      }
+
+      return data.map(frame => {
+        return {
+          ...frame,
+          fields: fieldsSorter(frame.fields),
+        };
+      });
+    };
+  },
+};
+
+const createFieldsSorter = (indexByName: Record<string, number>) => (fields: Field[]) => {
+  if (!Array.isArray(fields) || fields.length === 0) {
+    return fields;
+  }
+
+  return fields
+    .reduce((sortedFields, field) => {
+      sortedFields[getsertIndex(indexByName, field)] = field;
+      return sortedFields;
+    }, new Array<Field>(fields.length))
+    .filter(field => !!field);
+};
+
+const getsertIndex = (indexByName: Record<string, number>, field: Field): number => {
+  if (typeof indexByName[field.name] !== 'number') {
+    indexByName[field.name] = Object.keys(indexByName).length;
+  }
+  return indexByName[field.name];
+};
```

Comment:
not sure I understand this code, seems very fragile, if the field name is not found in IndexByName they will all overwrite the last index of Object.keys(indexByName).length  

Why not use a normal sort function and use indexByName as the sort by?

Reply:
or I guess indexByName[field.name] = Object.keys(indexByName).length; will increase the  Object.keys(indexByName).length by 1 each time so should not overwrite. 

getting null exceptions in getsertIndex  function, not sure why...

Reply:
hm.. was caused by empty options object

Reply:
The only reason I don't have a regular sort order is due to performance reasons. I wasn't sure if these parts of the code potentially can handle really large data sets.

But I am happy to change to a regular sort function. I am basically doing that in the editor.

Reply:
>  I wasn't sure if these parts of the code potentially can handle really large data sets.

yes, think about really large datasets... but they should grow in *length*, this deals with the list of fields -- that should be reasonably small numbers.  (think columns in a SQL table, not the rows)

The number of series?  It may be large, but i don't think it should ever be really really large.  Big numbers would be like the number of hosts in a cluster

Reply:
Super - then I will make this one a bit simpler.